crates/nodeapi/src/lib.rs

```diff
@@ -146,18 +146,32 @@ impl NodeApi {
 }
 
 #[allow(unused)]
-pub async fn node_info(api: web::types::State<NodeApi>) -> impl Responder {
+pub async fn node_info(api: web::types::State<Arc<Mutex<NodeApi>>>) -> impl Responder {
+    let data = match api.lock() {
+        Ok(guard) => guard,
+        Err(err) => {
+            return HttpResponse::InternalServerError()
+                .body(format!("Internal Server Error: {}", err));
+        }
+    };
     let response = serde_json::json!({
-        "node_name": api.node_name,
-        "node_version": api.node_version,
+        "node_name": data.node_name,
+        "node_version": data.node_version,
         "spec_version": "v0.0.1",
     });
     HttpResponse::Ok().json(&response)
 }
 
 #[allow(unused)]
-pub async fn health_check(api: web::types::State<NodeApi>) -> impl Responder {
-    let health = &api.health;
+pub async fn health_check(api: web::types::State<Arc<Mutex<NodeApi>>>) -> impl Responder {
```

Comment:
This should be private. Also, let's remove the `#[allow(unused)]`

Reply:
done